python/ray/data/datasource/parquet_datasource.py

```diff
@@ -199,6 +196,22 @@ def __init__(
                 ray.get_runtime_context().get_node_id(), soft=False
             )
 
+        paths, filesystem = _resolve_paths_and_filesystem(paths, filesystem)
+
+        # HACK: PyArrow's `ParquetDataset` errors if input paths contain non-parquet
+        # files. To avoid this, we expand the input paths with the default metadata
+        # provider and then apply the partition filter.
+        partition_filter = reader_args.pop("partition_filter", None)
+        if partition_filter is not None:
+            default_meta_provider = DefaultFileMetadataProvider()
```

Comment:
should we call `get_generic_metadata_provider()` instead?

Reply:
Ah, yeah, good call